File: stepik-java-api/src/main/java/org/stepik/api/queries/attempts/StepikAttemptsPostQuery.java
package org.stepik.api.queries.attempts;

import org.jetbrains.annotations.NotNull;
import org.stepik.api.actions.StepikAbstractAction;
import org.stepik.api.objects.attempts.Attempts;
import org.stepik.api.objects.attempts.AttemptsPost;
import org.stepik.api.queries.StepikAbstractPostQuery;
import org.stepik.api.urls.Urls;

/**
 * @author meanmail
 */
public class StepikAttemptsPostQuery extends StepikAbstractPostQuery<Attempts> {
    private final AttemptsPost attempts = new AttemptsPost();

    public StepikAttemptsPostQuery(@NotNull StepikAbstractAction stepikAction) {
        super(stepikAction, Attempts.class);
    }

    @NotNull
    public StepikAttemptsPostQuery step(int id) {
        attempts.getAttempt().setStep(id);
        return this;
    }

    @NotNull
    @Override
    protected String getBody() {
        return getJsonConverter().toJson(attempts);
    }

    @NotNull
    @Override
    protected String getUrl() {
        return Urls.ATTEMPTS;
    }
}

File: stepik-java-api/src/main/kotlin/org/stepik/api/queries/attempts/StepikAttemptsPostQuery.kt
package org.stepik.api.queries.attempts

import org.stepik.api.actions.StepikAbstractAction
import org.stepik.api.objects.attempts.Attempts
import org.stepik.api.objects.attempts.AttemptsPost
import org.stepik.api.queries.StepikAbstractPostQuery

class StepikAttemptsPostQuery(stepikAction: StepikAbstractAction) :
        StepikAbstractPostQuery<Attempts>(stepikAction, Attempts::class.java) {
    
    private val attempts = AttemptsPost()
    
    fun step(id: Long): StepikAttemptsPostQuery {
        attempts.attempt.step = id
        return this
    }
    
    override fun getBody(): String {
        return jsonConverter.toJson(attempts, false)
    }
    
    override fun getUrl(): String {
        return "${stepikAction.stepikApiClient.host}/api/attempts"
    }
}
